### DISCUSSION TITLE
How can I delete buffers from previous session when loading another

### DISCUSSION THREAD

**Discussion by FelipeSharkao** [olimorris/persisted.nvim#67] [Q&A]
Whenever I load a new session, the buffers from the previous keep in the buffer list. This is a fine default, but for me, it adds too much noise. I'm looking for a way to delete the buffers from the previous session, so they don't "leak". This is my burrent config:

```lua
require("persisted").setup({
    autosave = false,
})

vim.o.sessionoptions = "buffers,curdir,folds,globals,tabpages,winpos,winsize"

require("telescope").load_extension("persisted")

vim.keymap.set("n", "<Leader>p", function()
    vim.cmd("Telescope persisted")
end)

local function autosave()
    local files = vim.fn.readdir(vim.loop.cwd())

    for _, file in ipairs(files) do
        if file == ".git" or file == "package.json" or file == "Cargo.toml" then
            vim.cmd("SessionSave")
            break
        end
    end
end

vim.api.nvim_create_autocmd(
    { "VimLeavePre", "BufEnter", "BufDelete", "BufWinEnter", "DirChangedPre" },
    { pattern = "*", callback = autosave }
)

vim.api.nvim_create_autocmd("User", { pattern = "PersistedLoadPre", callback = autosave })
```

I've tried changing the `PersistedLoadPre` to

```lua
vim.api.nvim_create_autocmd("User", {
    pattern = "PersistedLoadPre",
    callback = function()
        autosave()
        vim.cmd("SessionStop")
        vim.cmd("silent! bufdo! noa bwipeout!")
    end
})
```

But then it stores the previous session with no buffers. I also tried recording which buffers should be deleted on `PersistedLoadPre`, and only clearing them on `PersistedLoadPost`, with no success.

**Comment by olimorris** [comment]
Have you tried the `PersistedTelescopeLoadPre` event? Assuming you're loading via Telescope

**Comment by FelipeSharkao** [comment]
Yes, I have tried. It stores an empty store same, as before

**Comment by FelipeSharkao** [comment]
It worked by using the `PersistedTelescopeLoadPre` event instead of `PersistedLoadPre` and only `VimLeavePre` in the autosave event. I still don't quite know why `noa` didn't work, but this is fine for now.

**Comment by dimas-cyriaco** [comment]
I think this should be the default behavior. It's weird that it keeps the previous session files open.

**Comment by olimorris** [comment]
> I think this should be the default behavior. It's weird that it keeps the previous session files open.

Weird is an odd choice of word.

My personal preference is to not force my preferences onto users of the plugin. Especially when a `vim.cmd` can achieve what you're looking for. Some users may wish to add buffers from another session into their current session. If I followed your suggestion, how would they opt out of that behaviour?
